dataland-frontend/tests/e2e/fixtures/custom_mocks/StoredDataRequestsFaker.ts

```diff
@@ -0,0 +1,131 @@
+import { RequestStatus, type StoredDataRequest } from "@clients/communitymanager";
+import { DataTypeEnum } from "@clients/backend";
+import { generateInt } from "@e2e/fixtures/common/NumberFixtures";
+import { generateReportingPeriod } from "@e2e/fixtures/common/ReportingPeriodFixtures";
+import { generateStoredDataRequestMessage } from "@e2e/fixtures/custom_mocks/StoredDataRequestMessageFaker";
+import { faker } from "@faker-js/faker";
+import { generateArray, pickOneElement } from "@e2e/fixtures/FixtureUtils";
+
+/**
+ * Creates a list of stored data requests
+ * @returns The list of request
+ */
+export function generateStoredDataRequests(): StoredDataRequest[] {
+  const storedDataRequests = [];
+  storedDataRequests.push(generateStoredDataRequest());
+  storedDataRequests.push(
+    manipulateFixtureToHaveDataType(
+      manipulateFixtureToHaveCompanyId(
+        manipulateFixtureToHaveReportingPeriod(
+          manipulateFixtureToHaveStatus(generateStoredDataRequest(), RequestStatus.Open),
+          "2021",
+        ),
+        "Mock-Company-Id",
+      ),
+      DataTypeEnum.Lksg,
+    ),
+  );
+  storedDataRequests.push(
+    manipulateFixtureToHaveDataType(
+      manipulateFixtureToHaveCompanyId(
+        manipulateFixtureToHaveReportingPeriod(
+          manipulateFixtureToHaveStatus(generateStoredDataRequest(), RequestStatus.Answered),
+          "2022",
+        ),
+        "Mock-Company-Id",
+      ),
+      DataTypeEnum.Lksg,
+    ),
+  );
+  storedDataRequests.push(
+    manipulateFixtureToHaveDataType(
+      manipulateFixtureToHaveCompanyId(
+        manipulateFixtureToHaveReportingPeriod(
+          manipulateFixtureToHaveStatus(generateStoredDataRequest(), RequestStatus.Answered),
+          "2024",
+        ),
+        "Mock-Company-Id",
+      ),
+      DataTypeEnum.Lksg,
+    ),
+  );
+  storedDataRequests.push(
+    manipulateFixtureToHaveDataType(
+      manipulateFixtureToHaveCompanyId(
+        manipulateFixtureToHaveReportingPeriod(
+          manipulateFixtureToHaveStatus(generateStoredDataRequest(), RequestStatus.Answered),
+          "1996",
+        ),
+        "550e8400-e29b-11d4-a716-446655440000",
+      ),
+      DataTypeEnum.EutaxonomyNonFinancials,
+    ),
+  );
+  return storedDataRequests;
+}
+
+/**
+ * Creates a default stored data request
+ * @returns The request
+ */
+export function generateStoredDataRequest(): StoredDataRequest {
+  const messageHistory = generateArray(() => generateStoredDataRequestMessage(), 1);
+  return {
+    dataRequestId: faker.string.uuid(),
+    userId: faker.string.uuid(),
+    creationTimestamp: generateInt(500),
+    dataType: pickOneElement(Object.values(DataTypeEnum)),
+    reportingPeriod: generateReportingPeriod(),
+    datalandCompanyId: faker.string.uuid(),
+    messageHistory: messageHistory,
+    lastModifiedDate: generateInt(500) + 500,
```

Comment:
the "500" should be saved in a constant imported form Constants.ts.
this is a magicNumber sonar issue